src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -15,18 +15,138 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.Arrays;
+import java.util.ArrayList;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
 
-/** Servlet to add user profile data to the Datastore */
-@WebServlet("/data")
+/**
+ * Servlet that provides information about a specific user, and allows setting a user's info.
+ * 
+ * <p>A User Entity consists of the following information: a user id, name, email, bio, and friends-list.
+ * These entities are stored in Datastore with their user id as the 'kind'.
+ *
+ * <p>TODO(#15): Add Blobstore Keys for each image that the user uploads.
+ */
+
+@WebServlet("/user-data")
 public class UserDataServlet extends HttpServlet {
+  private static final String DEFAULT_STRING = "";
+  private static final String USER_ENTITY = "User";
+  private static final String USER_BIO_PROPERTY = "bio";
+  private static final String USER_EMAIL_PROPERTY = "email";
+  private static final String USER_FOUND_PROPERTY = "user-found";
+  private static final String USER_FRIENDS_LIST_PROPERTY = "friends-list";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String USER_NAME_PROPERTY = "name";
+
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final Gson gson = new Gson();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
-    response.getWriter().println("Sample response.");
+    // Get the userId, and query Datastore for the corresponding entity
+    String userId = getStringParameter(request, USER_ID_PROPERTY, DEFAULT_STRING);
+    Entity userEntity = datastore.prepare(new Query(userId)).asSingleEntity();
+
+    ImmutableMap.Builder<String, Object> userDataBuilder = ImmutableMap.builder();
+    if (userEntity == null) {
+      // If a user entity was not found
+      userDataBuilder.put(USER_FOUND_PROPERTY, false);
+    }
+    else {
+      // If a user entity was found (a single entity)
+      // Get the user's information
+      userDataBuilder.put(USER_FOUND_PROPERTY, true);
```

Comment:
You can chain these (`put` returns the builder object):

```
userDataBuilder.put(USER_FOUND_PROPERTY, true)
    .put(USER_BIO_PROPERTY, (String) userEntity.getProperty(USER_BIO_PROPERTY))
    ...
    .put(...);
```

Reply:
Makes sense, I'll make the change.